bin/stress-test/src/store/mod.rs

```diff
@@ -195,9 +195,16 @@ pub async fn bench_check_nullifiers_by_prefix(
                 .iter()
                 .filter_map(|n| {
                     // private notes are filtered out because `n.details` is None
-                    Note::read_from_bytes(n.details.as_ref()?)
-                        .ok()
-                        .map(|n| u32::from(n.nullifier().prefix()))
+                    let inner_note = n.note.as_ref()?;
+                    let details =
+                        NoteDetails::read_from_bytes(inner_note.details.as_ref()?).unwrap();
+                    let (assets, recipient) = details.into_parts();
+                    let note = Note::new(
+                        assets,
+                        NoteMetadata::try_from(inner_note.clone().metadata?).ok()?,
+                        recipient,
+                    );
+                    Some(u32::from(note.nullifier().prefix()))
```

Comment:
We don't need to construct the full note to get the `Nullifier` - right? (we should be able to get it from `NoteDetails`). We could write it something like:
```rust
let detail_bytes = n.note.as_ref()?.details.as_ref()?;
let details = NoteDetails::read_from_bytes(detail_bytes).unwrap();
Some(u32::from(details.nullifier().prefix()))
```